File: BarrageForOC/BarrageForOC/SSLabel.m
//
//  SSLabel.m
//  BarrageForOC
//
//  Created by allthings_LuYD on 16/8/5.
//  Copyright © 2016年 scrum_snail. All rights reserved.
//

#import "SSLabel.h"
#define SSLableHeight 50
#define SSDuration 5
#define SSWidth [UIScreen mainScreen].bounds.size.width
#define SSHeight [UIScreen mainScreen].bounds.size.height
@implementation SSLabel

- (void)showInView:(UIView *)view withContent:(NSString *)content{
    self.frame = CGRectMake(SSWidth, arc4random_uniform((int)(SSHeight-100)), [self boundingRectWithContent:content], SSLableHeight);
    self.text = content;
    [view addSubview:self];
    [UIView animateWithDuration:SSDuration delay:0.0 options:UIViewAnimationOptionCurveLinear animations:^{
        CGRect rect = self.frame;
        rect.origin.x = -self.frame.size.width;
        self.frame = rect;
    } completion:^(BOOL finished) {
        [self removeFromSuperview];
    }];
}


- (CGFloat)boundingRectWithContent:(NSString *)content
{
    NSDictionary *attribute = @{NSFontAttributeName: self.font};

    CGSize retSize = [content boundingRectWithSize:CGSizeMake(MAXFLOAT, SSLableHeight)
                                          options:\
                      NSStringDrawingTruncatesLastVisibleLine |
                      NSStringDrawingUsesLineFragmentOrigin |
                      NSStringDrawingUsesFontLeading
                                       attributes:attribute
                                          context:nil].size;

    return retSize.width;
}

@end

File: BarrageForSwift/BarrageForSwift/SSLabel.swift
//
//  SSLabel.swift
//  BarrageForSwift
//
//  Created by allthings_LuYD on 16/8/5.
//  Copyright © 2016年 scrum_snail. All rights reserved.
//

import UIKit

class SSLabel: UILabel {
    let SSLabelHeight = 50
    let SSWidth  = UIScreen.main().bounds.size.width
    let SSHeight = UIScreen.main().bounds.size.height
    let SSDrution = 5
    func showInView(view:UIView,content:String) -> Void {

        self.frame = CGRect(x: SSWidth, y: CGFloat(arc4random_uniform(UInt32(Int(SSHeight - 100)))), width: self.boundingRectWithContent(content: content), height: CGFloat(SSLabelHeight))
        self.text = content
        view.addSubview(self)
        UIView.animate(withDuration: TimeInterval(SSDrution), delay: 0.0, options: UIViewAnimationOptions.curveLinear, animations: { 
            self.frame.origin.x = -self.frame.size.width
            }) { (true) in
                self.removeFromSuperview()
        }
    }

    func boundingRectWithContent(content:String) -> CGFloat {
        let option:NSStringDrawingOptions = [.usesLineFragmentOrigin,.usesFontLeading,.truncatesLastVisibleLine]
        var size = CGSize();
         size = content.boundingRect(with: CGSize(width: 9999,height: SSLabelHeight), options: option, attributes: [NSFontAttributeName : self.font], context: nil).size

//          size = content.boundingRect(with: CGSize(width: 9999,height: SSLabelHeight), options: OCUtils.stringDrawingOptions(), attributes: [NSFontAttributeName : self.font], context: nil).size
        return size.width
    }

}
